src/NServiceBus.Core/Pipeline/Incoming/TransportReceiveToPhysicalMessageProcessingConnector.cs

```diff
@@ -76,9 +75,10 @@ static void ConvertToPendingOperations(OutboxMessage deduplicationEntry, Pending
             }
         }
 
-        static List<TransportOperation> ConvertToOutboxOperations(IEnumerable<Transports.TransportOperation> operations)
+        static TransportOperation[] ConvertToOutboxOperations(Transports.TransportOperation[] operations)
```

Comment:
It seems odd to me that we have to take in an array of `TransportOperation` and return a new array of equal length of newly allocated `TransportOperation`s.